pennylane_lightning/_serialize.py

```diff
@@ -58,162 +40,210 @@
 }
 
 
-def _serialize_named_obs(ob, wires_map: dict, use_csingle: bool):
-    """Serializes a Named observable"""
-    named_obs = NamedObsC64 if use_csingle else NamedObsC128
-    wires = [wires_map[w] for w in ob.wires]
-    if ob.name == "Identity":
-        wires = wires[:1]
-    return named_obs(ob.name, wires)
-
-
-def _serialize_hermitian_ob(o, wires_map: dict, use_csingle: bool):
-    """Serializes a Hermitian observable"""
-    assert not isinstance(o, Tensor)
-
-    if use_csingle:
-        ctype = np.complex64
-        hermitian_obs = HermitianObsC64
-    else:
-        ctype = np.complex128
-        hermitian_obs = HermitianObsC128
-
-    wires = [wires_map[w] for w in o.wires]
-    return hermitian_obs(matrix(o).ravel().astype(ctype), wires)
-
+class _Serialize:
```

Comment:
```suggestion
class QuantumScriptSerializer:
```
I think we can call it something like the above. Also, since we are explicitly using this, I think we can get rid of the prepended underscore --- since this is used internally in the package, we don't need to follow the private conventions any more.